app/server/methods/applicationMethods.js

```diff
@@ -17,12 +17,39 @@ Meteor.methods({
   submitApplication: function(applicationId) {
     check(applicationId, String);
```

Comment:
it would be possible to submit another users application if you knew its ID number at this point. You could check that the user 'owns' the application that is being submitted. [but obv guessing IDs is hard]